### ISSUE TITLE
"thrconn undeclared" error during make, when I do cmake -DWITH_BINLOG=ON 

### ISSUE THREAD

**Issue by krlinus** [maxbube/mydumper#204]
/home/ec2-user/mydumper-src/mydumper-master/mydumper.c:1371:19: error: ‘thrconn’ undeclared (first use in this function); did you mean ‘conn’?                                                                                                                                           mysql_options(thrconn, MYSQL_READ_DEFAULT_FILE, defaults_file);                                                                                         ^~~~~~~                                                                                                                                   conn

**Comment by krlinus** [comment]
Sorry, not sure what I did wrong, but it is fine when I re-did everything.

**Comment by fredricj** [comment]
Fixed by #217 for older mysql versions. Still doesn't build against mysql 8 due to unrelated issues.

**Comment by cbtkris** [comment]
Issue still exists. Attempting compile with "cmake -DWITH_BINLOG=ON", I get the same reported error from make.  This is with latest 0.9.5 source code tarball against MySQL5.5 community repo on CentOS7 and the following:

$ cmake -DWITH_BINLOG=ON .
-- The C compiler identification is GNU 4.8.5
-- The CXX compiler identification is GNU 4.8.5
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Using mysql-config: /usr/bin/mysql_config
-- Found MySQL: /usr/include/mysql, /usr/lib64/mysql/libmysqlclient_r.so;/usr/lib64/libpthread.so;/usr/lib64/libm.so;/usr/lib64/libdl.so
-- Found ZLIB: /usr/lib64/libz.so (found version "1.2.7")
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.27.1")
-- checking for one of the modules 'glib-2.0'
-- checking for one of the modules 'gthread-2.0'
-- checking for module 'libpcre'
--   found libpcre, version 8.32
-- Found PCRE: /usr/include

CMake Warning at docs/CMakeLists.txt:9 (message):
  Unable to find Sphinx documentation generator


-- ------------------------------------------------
-- MYSQL_CONFIG = /usr/bin/mysql_config
-- CMAKE_INSTALL_PREFIX = /usr/local
-- BUILD_DOCS = ON
-- WITH_BINLOG = ON
-- RUN_CPPCHECK = OFF
-- Change a values with: cmake -D<Variable>=<Value>
-- ------------------------------------------------
--
-- Configuring done
-- Generating done
-- Build files have been written to: /home/centos/mydumper-0.9.5

$ make
Scanning dependencies of target mydumper
[ 11%] Building C object CMakeFiles/mydumper.dir/mydumper.c.o
/home/centos/mydumper-0.9.5/mydumper.c: In function ‘binlog_thread’:
/home/centos/mydumper-0.9.5/mydumper.c:1108:17: error: ‘thrconn’ undeclared (first use in this function)
   mysql_options(thrconn,MYSQL_READ_DEFAULT_FILE,defaults_file);
                 ^
/home/centos/mydumper-0.9.5/mydumper.c:1108:17: note: each undeclared identifier is reported only once for each function it appears in
make[2]: *** [CMakeFiles/mydumper.dir/mydumper.c.o] Error 1
make[1]: *** [CMakeFiles/mydumper.dir/all] Error 2
make: *** [all] Error 2

I am compiling from source because the CentOS 7 (EL7) RPM of mydumper does not have binlog support, which appears to prevent any binlog related info (master/slave status/replication position) from ever being written to the metadata file.

**Comment by cbtkris** [comment]
For the original poster, I just found that this same issue is reported under issue #61 by oracle1521. It was closed on Dec 27, 2017, listed as wontfix by repo owner.

My initial problem was seeing no replication related (show master status) output in my metadata file when I used EL7 rpm of mydumper.  This led me to compiling my own with feature I need (-DWITH_BINLOG=ON).  I emailed the owner in hopes of a reply, to help me clear any confusion and find out if I have any options that I'm not aware of.
